tests/backbeat/tests/api/metrics.js

```diff
@@ -1,191 +1,202 @@
-                    assert.strictEqual(keys[0], 'backlog');
-
-                    assert(body.backlog.description);
-                    assert(body.backlog.results);
-                    const resultKeys = Object.keys(body.backlog.results);
+                    const type = Object.keys(body)[0];
+                    const data = body[type];
+                    assert(data.description);
+                    assert(data.results);
+                    const resultKeys = Object.keys(data.results);
                     assert(resultKeys.includes('count'));
                     assert(resultKeys.includes('size'));
-                    next();
+                    return done()
                 });
-            }),
-    ], done));
+            });
+        });
+    });
+});
 
-    it('should get metrics when specifying valid site', done => series([
-        next => scalityUtils.putObject(srcBucket, key, Buffer.alloc(100), next),
-        next => scalityUtils.compareObjectsAWS(srcBucket, destBucket, key,
-            undefined, next),
-        next => makeGETRequest(
-            `/_/backbeat/api/metrics/crr/${destLocation}`,
-            (err, res) => {
-                assert.ifError(err);
-                assert.equal(res.statusCode, 200);
-                getResponseBody(res, (err, body) => {
-                    assert.ifError(err);
-                    const metricTypes = ['backlog', 'completions', 'throughput',
-                        'failures'];
-                    const keys = Object.keys(body);
-                    keys.forEach(key => {
-                        assert(metricTypes.includes(key));
-                        assert(body[key].description);
-                        assert(body[key].results);
-                        const resultKeys = Object.keys(body[key].results);
-                        assert(resultKeys.includes('count'));
-                        assert(resultKeys.includes('size'));
-                    });
+describe('Backbeat replication metrics data', function dF() {
+    this.timeout(REPLICATION_TIMEOUT);
+    const roleArn = 'arn:aws:iam::root:role/s3-replication-role';
+    const storageClass = `${destAWSLocation},${destAzureLocation}`;
+    const pathPrefix = '/_/backbeat/api/metrics/crr';
 
-                    next();
-                });
-            }),
+    beforeEach(done => series([
+        next => scalityUtils.createVersionedBucket(srcBucket, next),
+        next => scalityUtils.putBucketReplicationMultipleBackend(srcBucket,
+            'placeholder', roleArn, storageClass, next),
     ], done));
 
-    it('should get bucket-level throughput metrics', done => {
-        let versionId;
-        const destinationKey = `${srcBucket}/${key}`;
+    afterEach(done => parallel([
+        next => scalityUtils.deleteVersionedBucket(srcBucket, next),
+        next => awsUtils.deleteAllVersions(awsDestBucket, destKeyPrefix, next),
+        next => scalityUtils.deleteAllBlobs(destContainer, destKeyPrefix, next),
```

Comment:
I noticed a race condition here in another test run: since Azure does not have versioning, if `scalityUtils .deleteVersionedBucket` manages to replicate the deletion of the object version before `scalityUtils.deleteAllBlobs` tries to clean up the destination container, then a `BlobNotFound` error is thrown and `afterEach` fails. We can avoid this case by doing the cleanup in series, where we delete the blobs before deleting the source bucket.